movement/plots/occupancy.py

```diff
@@ -0,0 +1,134 @@
+
+from collections.abc import Hashable
+from typing import Any, Literal, TypeAlias
+
+import matplotlib.pyplot as plt
+import numpy as np
+import xarray as xr
+
+HistInfoKeys: TypeAlias = Literal["counts", "xedges", "yedges"]
+
+DEFAULT_HIST_ARGS = {"alpha": 1.0, "bins": 30, "cmap": "viridis"}
+
+
+def plot_occupancy(
+    da: xr.DataArray,
+    selection: dict[str, Hashable] | None = None,
+    ax: plt.Axes | None = None,
+    **kwargs: Any,
+) -> tuple[plt.Figure, plt.Axes, dict[HistInfoKeys, np.ndarray]]:
+    """Create a 2D histogram of the occupancy data given.
+
+    By default, the 0-indexed value along non-"time" and non-"space" dimensions
+    is plotted. The ``selection`` variable can be used to select different
+    coordinates along additional dimensions to plot instead.
+
+    Time-points whose corresponding spatial coordinates have NaN values
+    are ignored.
+
+    Histogram information is returned as the third output value (see Notes).
+
+    Parameters
+    ----------
+    da : xarray.DataArray
+        Spatial data to create histogram for. NaN values are dropped.
+    selection : dict[str, Hashable], optional
+        Mapping of dimension identifiers to the coordinate along that dimension
+        to plot. "time" and "space" dimensions are ignored. For example,
+        ``selection = {"individuals": "Bravo"}`` will create the occupancy
+        histogram for the individual "Bravo", instead of the occupancy
+        histogram for the 0-indexed entry on the ``"individuals"`` dimension.
+    ax : matplotlib.axes.Axes, optional
+        Axes object on which to draw the histogram. If not provided, a new
+        figure and axes are created and returned.
+    kwargs : Any
+        Keyword arguments passed to ``matplotlib.pyplot.hist2d``
+
+    Returns
+    -------
+    matplotlib.pyplot.Figure
+        Plot handle containing the rendered 2D histogram. If ``ax`` is
+        supplied, this will be the figure that ``ax`` belongs to.
+    matplotlib.axes.Axes
+        Axes on which the histogram was drawn. If ``ax`` was supplied,
+        the input will be directly modified and returned in this value.
+    dict[str, numpy.ndarray]
+        Information about the created histogram (see Notes).
+
+    Notes
+    -----
+    In instances where the counts or information about the histogram bins is
+    desired, the ``return_hist_info`` argument should be provided as ``True``.
+    This will force the function to return a second output value, which is a
+    dictionary containing the bin edges and bin counts that were used to create
+    the histogram.
+
+    For data with ``N`` time-points, the dictionary output has key-value pairs;
+    - ``xedges``, an ``(N+1,)`` ``numpy`` array specifying the bin edges in the
+    first spatial dimension.
+    - ``yedges``, same as ``xedges`` but for the second spatial dimension.
+    - ``counts``, an ``(N, N)`` ``numpy`` array with the count for each bin.
```

Comment:
Something here doesn't add up. Why would he shapes fo `xedges`, `yedges` and `counts` be related to the number of time-points `N`?
We are binning over space, not time.